Makefile

```diff
@@ -1,4 +1,7 @@
-all    :; dapp --use solc:0.6.12 build
+.PHONY: build
+
+all    :  build;
+build  :; DAPP_BUILD_OPTIMIZE=1 DAPP_BUILD_OPTIMIZE_RUNS=0 dapp --use solc:0.6.12 build
```

Comment:
`DAPP_BUILD_OPTIMIZE_RUNS=1`

Reply:
Actually, this is an LP oracle, so this should be a very high number I think.

Reply:
Actually, high runs did very little in testing (up to 1000 made no difference). And there was no difference between 1 and 0.

Reply:
In fact, I wonder if the `dapp` flag is broken...

Reply:
Actually it seems that the runs flag hasn't made into the latest "official" `dapp` yet, so no wonder it was doing nothing...

I'm going to go back to `SOLC_FLAGS` and see what happens there.

Reply:
So, using `SOLC_FLAGS` and `--legacy`, I get:
```
# runs | further gas savings on pass+poke
1K     | 40
10K    | 160
100K   | 208
1M     | 208
```
(It doesn't affect the gass for `pass` alone).

Btw, it's actually defaulting to 200 runs right now, regardless of `DAPP_BUILD_OPTIMIZE_RUNS`.

So, it looks like we can set the runs to 100K and max our gains...if we trust the optimizer that much :). The question is what's the best way to do it, since I don't like using the legacy features...maybe we should prod DappHub into releasing a version in which `DAPP_BUILD_OPTIMIZE_RUNS` is respected?

Reply:
Looks like DappHub updated the official release to respect `DAPP_BUILD_OPTIMIZE_RUNS`. I restructured the build mechanism to ensure that the `dapp` version is high enough to understand both new envvars.

The only open question is IMO "How many optimize runs is worthwhile?". Keep in mind there's an unknowable risk of optimizer bugs without bytecode-level formal verification. Higher runs probably add more risk, although not necessarily.